django/contrib/auth/tokens.py

```diff
@@ -128,5 +135,40 @@ def _now(self):
         # Used for mocking in tests
         return datetime.now()
 
+    def _xor_encrypt_decrypt(self, uid):
+        """
+        Performs XOR encryption/decryption on a uid to obfuscate
+        its value in the reset link.
+        This approach avoids adding a new dependency for encryption,
+        which is not natively part of Python's standard library.
+        The cypher key is a salted hash of the SECRET_KEY.
+        It is important that the cipher key is at least as long as the uid.
+        We use the SHA-512 hash algorithm to ensure a long enough cipher key (64 bytes)
+        to encrypt UUID4s (36 bytes) that might be used as primary key.
+        BigAutoField (the default primary key) is also supported since
+        it has a maximum size of 19 bytes. We cycle the key to also support the
+        unlikely scenario that the uid is longer than 64 bytes.
+        """
+        key = hashlib.sha512(force_bytes(f"{self.key_salt}{self.secret}")).digest()
+        uid_bytes = force_bytes(uid)
+        xor_ciphertext = bytes(
+            a ^ b for a, b in zip(uid_bytes, (key * (len(uid_bytes) // len(key) + 1)))
+        )
+        return xor_ciphertext
+
+    def encrypt_uid(self, uid):
+        """
+        Returns a XOR-encrypted user id for use in the password reset mechanism.
```

Comment:
Docstrings should follow PEP-8 and PEP-257, specifically verbs should be written as a command. An alternative proposal would be:
```suggestion
        XOR-encrypt the provided user ID and return its base64-encoded representation.
```